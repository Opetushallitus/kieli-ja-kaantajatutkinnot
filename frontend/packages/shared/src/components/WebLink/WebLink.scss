.weblink {
    .icon {
        vertical-align: -0.5em;
        padding: 0 0.2em;
    }
}
