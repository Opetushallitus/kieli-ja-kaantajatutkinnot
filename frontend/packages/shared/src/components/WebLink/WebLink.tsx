import { Link } from '@mui/material';
import React, { FC } from 'react';

export interface WebLinkProps {
  href: string;
  label: string;
  target?: string;
  startIcon?: React.ReactNode;
  endIcon?: React.ReactNode;
}

export const WebLink: FC<WebLinkProps> = ({
  href,
  label,
  target = '_blank',
  startIcon,
  endIcon,
}) => {
  const link = (
    <Link sx={{ fontWeight: 400 }} href={href} target={target}>
      {label}
    </Link>
  );

  if (!startIcon && !endIcon) {
    return <>{link}</>;
  }

  return (
    <div className="columns gapped-xxs">
      {startIcon && startIcon}
      {link}
      {endIcon && endIcon}
    </div>
  );
};
