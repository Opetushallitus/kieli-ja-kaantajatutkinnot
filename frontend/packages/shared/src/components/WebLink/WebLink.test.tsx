import DownloadIcon from '@mui/icons-material/DownloadOutlined';
import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import renderer from 'react-test-renderer';

import { WebLink } from './WebLink';

describe('WebLink', () => {
  it('should render correctly without icons', () => {
    const tree = renderer
      .create(<WebLink href="http://localhost:4000" label="Localhost" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with start and end icons', () => {
    const tree = renderer
      .create(
        <WebLink
          href="http://localhost:4000"
          label="Localhost"
          target=""
          startIcon={<DownloadIcon />}
          endIcon={<OpenInNewIcon />}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
