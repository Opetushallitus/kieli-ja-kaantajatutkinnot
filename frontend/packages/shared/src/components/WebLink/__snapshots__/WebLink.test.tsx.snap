// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebLink should render correctly with start and end icons 1`] = `
<span
  className="weblink"
>
  <span
    className="icon start-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="DownloadOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 9h-4V3H9v6H5l7 7zm-8 2V5h2v6h1.17L12 13.17 9.83 11zm-6 7h14v2H5z"
      />
    </svg>
  </span>
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-ja4uiq-MuiTypography-root-MuiLink-root"
    href="http://localhost:4000"
    onBlur={[Function]}
    onFocus={[Function]}
    target=""
  >
    Localhost
  </a>
  <span
    className="icon end-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="OpenInNewIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
      />
    </svg>
  </span>
</span>
`;

exports[`WebLink should render correctly without icons 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-ja4uiq-MuiTypography-root-MuiLink-root"
  href="http://localhost:4000"
  onBlur={[Function]}
  onFocus={[Function]}
  target="_blank"
>
  Localhost
</a>
`;
